File: features/structures/enumerations.dart
//Dart only supports structs with fields of types int, double, or pointer
enum Rank {
    ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN,
    JACK, QUEEN, KING
}

rankDescription(Rank r) {
  switch(r) {
      case Rank.ACE:
          return "ace";
      case Rank.JACK:
          return "jack";
      case Rank.QUEEN:
          return "queen";
      case Rank.KING:
          return "king";
      default:
          return (r.index + 1).toString();
  }
}


enum Suit {
    SPADES, HEARTS, DIAMONDS, CLUBS
}

suitDescription(Suit s) {
  switch(s) {
      case Suit.SPADES : return "spades";
      case Suit.HEARTS : return "hearts";
      case Suit.DIAMONDS : "diamonds";
      case Suit.CLUBS : "clubs";
  }
}

void main() {
	Rank ace = Rank.ACE;
	Suit hearts = Suit.HEARTS;
	rankDescription(ace);
	suitDescription(hearts);
}

File: features/structures/enumerations.kt

enum class Rank {
    ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN,
    JACK, QUEEN, KING
}

fun rankDescription(r : Rank) : String {
    return when(r) {
        Rank.ACE ->  "ace"
        Rank.JACK ->  "jack"
        Rank.QUEEN ->  "queen"
        Rank.KING ->  "king"
        else -> (r.ordinal + 1).toString()
    }
}



enum class Suit {
    SPADES, HEARTS, DIAMONDS, CLUBS
}

fun suitDescription(s : Suit) : String {
    return when(s) {
        Suit.SPADES -> "spades"
        Suit.HEARTS -> "hearts"
        Suit.DIAMONDS -> "diamonds"
        Suit.CLUBS -> "clubs"
    }
}


fun main() {
    val ace = Rank.ACE
    val hearts = Suit.HEARTS
    rankDescription(ace)
    suitDescription(hearts)
}

